File: s2/1138/1138.cpp
#include<iostream>
#include<vector>
using namespace std;

int main()
{
	int n;
	cin >> n;
	int *people = new int [n];
	for(int i=0; i<n; i++)
		cin >> people[i];
	vector<int> order = {n};
	vector<int>::iterator iter;
	for(int i=n-1; i>0; i--)
	{
		int count = people[i-1];
		bool check_input = false;
		for(iter = order.begin(); iter != order.end(); iter ++)
		{
			if(count == 0)
			{
				order.insert(iter, i);
				check_input = true;
				break;
			}
			if(*iter > i)
				count --;
		}
		if(!check_input)
			order.insert(order.end(), i);
	}
	for(iter = order.begin(); iter != order.end(); iter++)
		cout << *iter << " ";
	cout << "\n";
	delete[] people;
	return 0;
}

File: s2/1138/1138.py
n = int(input())
people = list(map(int, input().split()))
order = [n]
for i in reversed(range(1, n)):
    count = people[i-1]
    for j in range(len(order)+1):
        if count == 0:
            order.insert(j, i)
            break
        if order[j] > i:
            count -= 1
for i in order:
    print(i, end = ' ')
print()
